inf-clojure.el

```diff
@@ -510,10 +519,11 @@ Fallback to `default-directory.' if not within a project."
 
 (defun inf-clojure-project-type ()
   "Determine the type, either leiningen or boot of the current project."
-  (let ((default-directory (inf-clojure-project-root)))
-    (cond ((file-exists-p "project.clj") "lein")
-          ((file-exists-p "build.boot") "boot")
-          (t nil))))
```

Comment:
We should be returning `"generic"` if the project type can't be determined, so we can leverage the `defcustom` - if it's nil it does nothing, otherwise it takes precedence over this.